### ISSUE TITLE
Why do you use CrossEntropyLoss on softmax output?

### ISSUE THREAD

**Issue by odats** [Sayan98/pytorch-segnet#5]
Model:
`x_softmax = F.softmax(x_00d, dim=1)`

Train:
https://github.com/say4n/pytorch-segnet/blob/be538aecd677b703e0cd9d803b51e33c06fa8a27/src/train.py#L73

https://pytorch.org/docs/stable/nn.html#torch.nn.CrossEntropyLoss
This criterion combines nn.LogSoftmax() and nn.NLLLoss() in one single class.

**Comment by say4n** [comment]
Interesting find!
It should be `predicted_tensor` instead, thanks! 😄
